internal/venice-client-common/src/main/java/com/linkedin/venice/stats/VeniceOpenTelemetryMetricsRepository.java

```diff
@@ -221,14 +228,61 @@ public Object createInstrument(MetricEntity metricEntity) {
     }
   }
 
+  public String getDimensionName(VeniceMetricsDimensions dimension) {
+    return dimension.getDimensionName(getMetricFormat());
+  }
+
+  private void validateDimensionValuesAndThrow(
+      MetricEntity metricEntity,
+      VeniceMetricsDimensions dimension,
+      String dimensionValue) {
+    if (dimensionValue == null || dimensionValue.isEmpty()) {
+      String errorLog = "Dimension value cannot be null or empty for key: " + dimension + " for metric: "
+          + metricEntity.getMetricName();
+      if (!REDUNDANT_LOG_FILTER.isRedundantLog(errorLog)) {
+        LOGGER.error(errorLog);
+      }
+      throw new VeniceException(errorLog);
+    }
+  }
+
+  public Attributes createAttributes(
+      MetricEntity metricEntity,
+      Map<VeniceMetricsDimensions, String> commonDimensionsMap,
+      Map<VeniceMetricsDimensions, String> additionalDimensionsMap) {
+    AttributesBuilder attributesBuilder = Attributes.builder();
+
+    // add common dimensions
+    for (Map.Entry<VeniceMetricsDimensions, String> entry: commonDimensionsMap.entrySet()) {
+      VeniceMetricsDimensions dimension = entry.getKey();
+      String dimensionValue = entry.getValue();
+      validateDimensionValuesAndThrow(metricEntity, dimension, dimensionValue);
+      attributesBuilder.put(getDimensionName(dimension), dimensionValue);
+    }
+
+    // add additional dimensions
+    for (Map.Entry<VeniceMetricsDimensions, String> entry: additionalDimensionsMap.entrySet()) {
+      VeniceMetricsDimensions dimension = entry.getKey();
+      String dimensionValue = entry.getValue();
+      validateDimensionValuesAndThrow(metricEntity, dimension, dimensionValue);
+      attributesBuilder.put(getDimensionName(dimension), dimensionValue);
+    }
```

Comment:
Nit, some duplicate code with the above block. Can we extract them into a small helper function?